deltachat-ffi/deltachat.h

```diff
@@ -441,7 +441,53 @@ char*           dc_get_info                  (dc_context_t* context);
  */
 char*           dc_get_oauth2_url            (dc_context_t* context, const char* addr, const char* redirect_uri);
 
+/**
+ * Get the provider information from our provider overview as json object.
+ */
+char*           dc_get_json_provider_info_from_email  (char* email);
+
+typedef struct _dc_provider  dc_provider_t;
+
+/**
+ * Returns a provider for an email address or null if we have no information about it.
+ * The result can be passed to dc_provider_get_*.
+ */
```

Comment:
How do i free this structure?  I'm going to stop writing this, but needs to be done everywyere.

Also, should point out the error value: NULL (which is the dc convention)

Reply:
the provider structures in the rust package (provider-overview) are static so I believe you can't free it.

Reply:
Cool.

Second guessing the long future (always a bad idea) I wonder if it's reasonable to provide the dc_provider_unref() function anyway which does nothing as it allows more flexibility to do things to the implementation later.  It is also more symmetrical with all the existing deltachat APIs.

Reply:
@flub What would be the advantage over adding this later?

Reply:
The clients would already be calling it, otherwise if you introduce it later all the clients would probably leak memory.

Reply:
If you say so :wink: If you think that it makes sense, I'll add it. How?

Reply:
I tried to do this but didn't work for some reason.  let's live without

Reply:
done now

Reply:
@flub: great, having a unref function makes much sense wrt consistence to the other api.

maybe we should also pass the context, may be needed eg. for logging, not sure. so, the way usage would be:

```
dc_provider_t* provider = dc_get_provider_by_domain(context, domain);

char* name = dc_provider_get_overview_name(provider);
...
dc_provider_unref(provider);
```

this is the same most other structs are generated.

but i would also be fine with the existing dc_provider_new_from_*() way; this is similar to the way message-objects are created.